### ISSUE TITLE
Distributed training takes more time than non distributed

### ISSUE THREAD

**Issue by piyushshri** [Russell91/TensorBox#81]
I have referred the [replicated training deep-MNIST example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dist_test/python/mnist_replica.py#L258) to distribute the training in TensorBox over multiple workers. The deep mnist example takes less time in distributed mode than non distributed, however, this is not the case with TensorBox.
The distributed version is taking more time to complete the training (approx 2 seconds per image) as compared to non distributed (300 ms per image) on my GPU cluster.
I am printing the global step in both [TensorBox](https://gist.github.com/piyushshri/5894d98e8e6089ad0e441ddcd8ca6433#file-distributed-tensorbox-trainer-py-L550) and [MNIST](https://gist.github.com/piyushshri/5894d98e8e6089ad0e441ddcd8ca6433#file-distributed-mnist-trainer-py-L158) to monitor the time taken per step with batch size 1.

**Comment by Russell91** [comment]
Are you sure the distributed mode is running no multiple gpus? Maybe use nvidia-smi during training and make sure GPU utilization is good.

**Comment by piyushshri** [comment]
Hi @Russell91 . Yes I have done multiple experiments to make sure that all the resources are being used:
1. The train loss reduces cumulatively on all the workers. If worker 1 started with loss 4.0, no other worker starts at 4.0.
2. The number of displaying in Tensorboard is the sum of number of iterations on each worker.
3. The global_step variable is being shared as expected.

I am attaching screenshots of `nvidia-smi` on two of the workers. Can you review the code for if there's a better approach to do it?
![screenshot from 2016-10-03 15 58 06](https://cloud.githubusercontent.com/assets/8086083/19034864/a7e8e4f4-8983-11e6-8918-0d3024b45054.png)
![worker1](https://cloud.githubusercontent.com/assets/8086083/19034865/a7e9f0ec-8983-11e6-9830-c5fe4b5c3bfa.png)

**Comment by Russell91** [comment]
Hmm, so you have 2 gpus on 2 different physical machines? If you aren't using GPU2GPU mem transfer through the pci bus, that can really slow things down. Network transfers are usually <300 MB/s, which is pretty slow for sharing dloss/dW for multi-gigabyte weight matrices.

**Comment by piyushshri** [comment]
@Russell91 we have multiple (two in this example) GPUs on multiple machines i.e. horizontal scaling. The network transfer rate being utilized is 30MB/s. What is the recommended network architecture we should use for the multi-gigabyte weight matrices you have mentioned.
We are planning to use [Amazon EC2 - P2 Instance - 8x.large](https://aws.amazon.com/ec2/instance-types/p2/) which has 10 Gbps network bandwidth. Any recommendations?
